File: chattools/src/main/java/aytackydln/chattools/telegram/dto/response/UpdateMessage.java
package aytackydln.chattools.telegram.dto.response;

import aytackydln.chattools.telegram.exception.TelegramException;
import com.fasterxml.jackson.annotation.JsonUnwrapped;
import com.fasterxml.jackson.databind.PropertyNamingStrategy;
import com.fasterxml.jackson.databind.annotation.JsonNaming;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.ToString;

@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
@Data
@AllArgsConstructor
@ToString
public class UpdateMessage implements TelegramResponse {
    private long messageId;

    private SendMessage sendMessage;

    @JsonUnwrapped
    public SendMessage getSendMessage() {
        return sendMessage;
    }

    @Override
    public final String getMethod() {
        return "editMessageText";
    }

    @Override
    public boolean isLimited() {
        return false;
    }

    @Override
    public TelegramResponse onError(TelegramException e) {
        return null;
    }

    @Override
    public void preSend() {
        //doesn't need
    }
}

File: chattools/src/main/kotlin/aytackydln/chattools/telegram/dto/response/UpdateMessage.kt
package aytackydln.chattools.telegram.dto.response

import aytackydln.chattools.telegram.exception.TelegramException
import com.fasterxml.jackson.annotation.JsonUnwrapped
import com.fasterxml.jackson.databind.PropertyNamingStrategies
import com.fasterxml.jackson.databind.PropertyNamingStrategy
import com.fasterxml.jackson.databind.annotation.JsonNaming
import lombok.AllArgsConstructor
import lombok.Data
import lombok.ToString

@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy::class)
@ToString
data class UpdateMessage(
    val messageId: Long = 0,
    @get:JsonUnwrapped val sendMessage: SendMessage,
) : TelegramResponse {

    override val method: String
        get() = "editMessageText"

    override val isLimited: Boolean
        get() = false

    override fun onError(e: TelegramException): TelegramResponse? {
        return null
    }

    override fun preSend() {
        //doesn't need
    }
}
